### DISCUSSION TITLE
What maintenance tasks should be done to harden the coolify mains / workers

### DISCUSSION THREAD

**Discussion by xXKynarethXx** [coollabsio/coolify#3043] [Q&A]
When connecting a new plain server, what tasks should be done to harden new servers?

- install / configure / enable  firewall (e.g. ufw)
- fail2ban
- Automatic System and Docker Updates / upgrades (maybe failure mails)
- Move default ssh port (also to not cause troubles with like git services)
- Configure and use new user as Root User and maybe disable Root Login?
- other recommendations?

Is maybe part of that already included in the setup/validation process of the new nodes / in the install script of coolify?

**Comment by visualcookie** [comment]
i wonder that too. i've a vaultwarden running and want to harden it with fail2ban. before i had a custom traefik configuration and ran all my docker containers manually, but i wanted to give coolify a shot but can't find any informations about securing containers or even backing up chosen containers with borgmatic. hope i'm not alone with this, but i would love to get some clarifications on this topic.

**Comment by LEstradioto** [comment]
I was looking for a simple blog post, this seems nice https://ivansalloum.com/linux-server-security-and-hardening/

**Comment by weewoo-meow** [comment]
Docker and ufw don't play well together out of the box and will ignore the ufw rules because routing is handled directly at iptables. See this: https://github.com/chaifeng/ufw-docker
